src/aiida/cmdline/utils/common.py

```diff
@@ -486,13 +486,17 @@ def build_entries(ports):
         echo.echo(style('\nExit codes that invalidate the cache are marked in bold red.\n', italic=True))
 
 
-def generate_validate_output_file(
-    output_file: Path | None, entity_label: str, appendix: str = '', overwrite: bool = False
+def validate_output_filename(
+    output_file: Path | str | None,
+    overwrite: bool = False,
 ):
-    """Generate default output filename for `Code`/`Computer` export and validate."""
+    """Validate output filename."""
 
     if output_file is None:
-        output_file = Path(f'{entity_label}{appendix}.yml')
+        raise ValueError('Output filename must be passed for validation.')
```

Comment:
nitpick: shouldn't this technically not be a `TypeError`?

Reply:
I initially used `ValueError` following the logic that the wrong _value_ is passed to the function, i.e., no _value_. But I agree, now that `None` is not accepted anymore, `TypeError` seems more appropriate. Cheers!